### PR TITLE
Return value stack constraint

### PR DESCRIPTION
**Pull request by saberduck**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java [hunk 1]
```diff
@@ -748,11 +756,11 @@ private boolean handleMethodInvocation(Instruction instruction) {
...
+          Type exceptionType = yield.exceptionType();
           yield.statesAfterInvocation(
             arguments, Collections.emptyList(), pop.state, () -> constraintManager.createExceptionalSymbolicValue(exceptionType)).forEach(ps -> {
-              ps.storeExitValue();
-              enqueueExceptionHandlers(exceptionType, ps);
+            ps.storeExitValue();
+            enqueueExceptionHandlers(exceptionType, ps);
           });
         });
       // FIXME : empty yields here should not happen, for now act as if behavior was not resolved.
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java [hunk 2]
```diff
@@ -757,11 +756,11 @@ private boolean handleMethodInvocation(Instruction instruction) {
       methodInvokedBehavior
         .exceptionalPathYields()
         .forEach(yield -> {
-          org.sonar.plugins.java.api.semantic.Type exceptionType = yield.exceptionType();
+          Type exceptionType = yield.exceptionType();
           yield.statesAfterInvocation(
             arguments, Collections.emptyList(), pop.state, () -> constraintManager.createExceptionalSymbolicValue(exceptionType)).forEach(ps -> {
-              ps.storeExitValue();
-              enqueueExceptionHandlers(exceptionType, ps);
+            ps.storeExitValue();
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java]: Not sure about this fix. To me this previous indentation was correct

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BytecodeEGWalker.java#L767](https://github.com/SonarSource/sonar-java/blob/dede4b519692ac87900a3298cfaf5919c9400911/java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java#L767): Take the required action to fix the issue indicated by this comment. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)
1. ![MINOR][MINOR] [BytecodeEGWalker.java#L760](https://github.com/SonarSource/sonar-java/blob/dede4b519692ac87900a3298cfaf5919c9400911/java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java#L760): Replace this fully qualified name with "Type" [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1942)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BytecodeEGWalker.java#L767](https://github.com/SonarSource/sonar-java/blob/5e416b4e34cde911554921634fc1d9b0433c21b2/java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java#L767): Take the required action to fix the issue indicated by this comment. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)
1. ![MINOR][MINOR] [BytecodeEGWalker.java#L760](https://github.com/SonarSource/sonar-java/blob/5e416b4e34cde911554921634fc1d9b0433c21b2/java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java#L760): Replace this fully qualified name with "Type" [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1942)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**